airbyte-workers/src/main/java/io/airbyte/workers/WorkerApp.java

```diff
@@ -134,72 +134,36 @@ public void start() {
+    if (configs.shouldRunGetSpecWorkflows()) {
+      registerGetSpec(factory);
+    }
 
-    final Worker checkConnectionWorker =
-        factory.newWorker(TemporalJobType.CHECK_CONNECTION.name(), getWorkerOptions(maxWorkers.getMaxCheckWorkers()));
-    checkConnectionWorker.registerWorkflowImplementationTypes(CheckConnectionWorkflowImpl.class);
-    checkConnectionWorker
-        .registerActivitiesImplementations(
-            new CheckConnectionActivityImpl(checkWorkerConfigs, checkProcessFactory, secretsHydrator, workspaceRoot, workerEnvironment, logConfigs,
-                jobPersistence, airbyteVersion));
+    if (configs.shouldRunCheckConnectionWorkflows()) {
+      registerCheckConnection(factory);
+    }
 
-    final Worker discoverWorker = factory.newWorker(TemporalJobType.DISCOVER_SCHEMA.name(), getWorkerOptions(maxWorkers.getMaxDiscoverWorkers()));
-    discoverWorker.registerWorkflowImplementationTypes(DiscoverCatalogWorkflowImpl.class);
-    discoverWorker
-        .registerActivitiesImplementations(
-            new DiscoverCatalogActivityImpl(discoverWorkerConfigs, discoverProcessFactory, secretsHydrator, workspaceRoot, workerEnvironment,
-                logConfigs,
-                jobPersistence, airbyteVersion));
+    if (configs.shouldRunDiscoverWorkflows()) {
+      registerDiscover(factory);
+    }
 
-    final NormalizationActivityImpl normalizationActivity =
-        new NormalizationActivityImpl(
-            containerOrchestratorConfig,
-            defaultWorkerConfigs,
-            defaultProcessFactory,
-            secretsHydrator,
-            workspaceRoot,
-            workerEnvironment,
-            logConfigs,
-            jobPersistence,
-            airbyteVersion);
-    final DbtTransformationActivityImpl dbtTransformationActivity =
-        new DbtTransformationActivityImpl(
-            containerOrchestratorConfig,
-            defaultWorkerConfigs,
-            defaultProcessFactory,
-            secretsHydrator,
-            workspaceRoot,
-            workerEnvironment,
-            logConfigs,
-            jobPersistence,
-            airbyteVersion);
-    new PersistStateActivityImpl(workspaceRoot, configRepository);
-    final PersistStateActivityImpl persistStateActivity = new PersistStateActivityImpl(workspaceRoot, configRepository);
-    final Worker syncWorker = factory.newWorker(TemporalJobType.SYNC.name(), getWorkerOptions(maxWorkers.getMaxSyncWorkers()));
-    final ReplicationActivityImpl replicationActivity = getReplicationActivityImpl(
-        containerOrchestratorConfig,
-        replicationWorkerConfigs,
-        replicationProcessFactory,
-        secretsHydrator,
-        workspaceRoot,
-        workerEnvironment,
-        logConfigs,
-        jobPersistence,
-        airbyteVersion);
-    syncWorker.registerWorkflowImplementationTypes(SyncWorkflowImpl.class);
+    if (configs.shouldRunSyncWorkflows() || configs.shouldRunConnectionManagerWorkflows()) {
```

Comment:
Is this outer conditional redundant since both conditions are checked individually inside anyways?

Reply:
There was a reason for this previously. Unnecessary now.